symengine/polynomial.cpp

```diff
@@ -409,19 +410,16 @@ std::size_t UnivariatePolynomial::__hash__() const
     std::size_t seed = UNIVARIATEPOLYNOMIAL;
 
     seed += hash_string(this->var_->get_name());
-    for (const auto &it : this->dict_) {
-        std::size_t temp = UNIVARIATEPOLYNOMIAL;
-        hash_combine<unsigned int>(temp, it.first);
-        hash_combine<Basic>(temp, *(it.second.get_basic()));
-        seed += temp;
-    }
+    seed += dict_.__hash__();
```

Comment:
So what should we do with UnivariateExprPolynomial's hash in UnivariateSeries::__hash__?